### PR TITLE
Substrate companion: Authority discovery multiple peer ids

### PR DESCRIPTION
**Pull request by bkchr**
Authority discovery before had a fixed mapping from `PeerId` to
`AuthorityId`. This wasn't correct, as a `PeerId` can actually map to
multiple `AuthorityId`s. The linked Substrate pr fixes this.

https://github.com/paritytech/substrate/pull/10259

skip check-dependent-cumulus

### CODE REVIEW

File: node/network/availability-distribution/src/requester/mod.rs [hunk 1]
```diff
@@ -159,7 +159,7 @@ impl Requester {
 				// Just book keeping - we are already requesting that chunk:
 				{
 					e.get_mut().add_leaf(leaf);
-				},
+				}
```
Comment (bkchr) [inline node/network/availability-distribution/src/requester/mod.rs:162]:
```suggestion
				},
```

File: node/network/collator-protocol/src/validator_side/mod.rs [hunk 2]
```diff
@@ -1423,7 +1423,7 @@ async fn poll_collation_response(
 				);
 
 				CollationFetchResult::Error(COST_WRONG_PARA)
-			},
+			}
```
Comment (bkchr) [inline node/network/collator-protocol/src/validator_side/mod.rs:1426]:
```suggestion
			},
```

File: node/network/statement-distribution/src/tests.rs [hunk 3]
```diff
@@ -1814,7 +1814,7 @@ fn peer_cant_flood_with_large_statements() {
 					if p == peer_a && r == COST_APPARENT_FLOOD =>
 				{
 					punished = true;
-				},
+				}
```
Comment (bkchr) [inline node/network/statement-distribution/src/tests.rs:1817]:
```suggestion
				},
```

File: node/network/statement-distribution/src/lib.rs [hunk 4]
```diff
@@ -1466,14 +1466,18 @@ async fn handle_network_update(
 					maybe_authority: maybe_authority.clone(),
 				},
 			);
-			if let Some(authority) = maybe_authority {
-				authorities.insert(authority, peer);
+			if let Some(authority_ids) = maybe_authority {
+				authority_ids.into_iter().for_each(|a| {
+					authorities.insert(a, peer);
```
Comment (ordian) [inline node/network/statement-distribution/src/lib.rs:1471]: The code here and the signature of `authorities: &mut HashMap<AuthorityDiscoveryId, PeerId>` still assumes one `PeerId` per `AuthorityDiscoveryId`. You've tweaked that in collator_side and gossip-support (only used for logs), should we also change it here?

Comment (ordian) [reply node/network/statement-distribution/src/lib.rs:1471]: Actually, it's not critical here, only used for prioritizing propagation to our group (priority_peers).

**Review by ordian — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Pull request merged**